api/src/methods/new_payload.rs

```diff
@@ -29,21 +30,15 @@ async fn inner_execute_v3(
     execution_payload: ExecutionPayloadV3,
     expected_blob_versioned_hashes: Vec<B256>,
     parent_beacon_block_root: B256,
-    state_channel: mpsc::Sender<StateMessage>,
+    app: &Arc<RwLock<Application<impl Dependencies>>>,
 ) -> Result<PayloadStatusV1, JsonRpcError> {
     // Spec: https://github.com/ethereum/execution-apis/blob/main/src/engine/cancun.md#specification
 
-    let (tx, rx) = oneshot::channel();
-    let msg = Query::GetPayloadByBlockHash {
-        block_hash: execution_payload.block_hash,
-        response_channel: tx,
-    }
-    .into();
-    state_channel.send(msg).await.map_err(access_state_error)?;
-    let maybe_response = rx.await.map_err(access_state_error)?;
-
     // TODO: in theory we should start syncing to learn about this block hash.
```

Comment:
Nit: this comment has moved a few lines further from the error handling it is referencing. Maybe it makes sense to put it inside the `ok_or` now.